src/play.jl

```diff
@@ -1,50 +1,132 @@
 """
-    GamePool
+    GamePool{N,S}
 
-A struct that defines a Wordle-like game.
+A struct that defines a Wordle-like game with targets of length `N`.
+`S <: Unsigned` is the smallest unsigned integer type that can store values in the
+range `0:(3 ^ N - 1)`, evaluated as `Wordlegames.scoretype(N)`
+
+The fields are:
+
+- `targetpool`: the target pool as a `Vector{NTuple{N,Char}}`
+- `guesspool`: a `Vector{NTuple{N,Char}}` of potential guesses (can be the same as `targets`)
+- `allscores`: a `Matrix{S}` where `allscores[i,j] = score(guesspool[j], targets[i])`
+- `active`: a `BitVector`. The active target pool is `targets[active]`.
+- `counts`: `Vector{Int}` of length `3 ^ N` in which bin counts are accumulated
```

Comment:
why is this `Int` instead of `S`?

Reply:
`S` is the smallest `Unsigned` type that can represent a score.  The maximum value in `counts` is the length of the target pool, which can be bigger than the maximum score.

Reply:
Ah, yeah makes sense :smile: